l1-contracts/src/core/libraries/rollup/STFLib.sol

```diff
@@ -95,17 +205,48 @@ library STFLib {
     emit IRollupCore.PrunedPending(proven, pending);
   }
 
+  /**
+   * @notice Calculates the size of the circular storage buffer for temporary block logs
+   * @dev The roundabout size determines how many blocks can be stored in the circular buffer
+   *      before older entries are overwritten. The size is calculated as:
+   *
+   *      roundaboutSize = maxPrunableBlocks() + 1
+   *
+   *      Where maxPrunableBlocks() = epochDuration * (proofSubmissionEpochs + 1)
+   *
+   *      This ensures that:
+   *      - All blocks within the proof submission window remain accessible
+   *      - At least one additional slot is available for the next block
```

Comment:
It is not to ensure that the next is available, but to ensure that at least one proven block is available (the last proven) so there is a "trusted anchor"